homework9/src/pages/Apply/Apply.jsx

```diff
@@ -0,0 +1,194 @@
+            <div className="apply--content__left--1">
+              <h3>Apply to desired course right away!</h3>
+              <p>
+                Select desired course and group, leave us your contact
+                information and we’ll reach out to you for confirmation and
+                letting you know about further steps
+              </p>
+            </div>
+            <div className="apply--content__left--2">
+              <InputApply
+                title="Course"
+                type="select"
+                options={newCourse}
+                onChange={(value) => {
+                  setSelectedCourse(value);
+                  setFormData((it) => {
+                    return { ...it, courseId: value.id };
+                  });
+                }}
+                isDisabled={!courses}
+              />
+
+              <InputApply
+                title="Group"
+                type="select"
+                options={courseGroup}
+                onChange={(value) =>
+                  setFormData((it) => {
+                    return { ...it, groupId: value.id };
+                  })
+                }
+                setFormData={setFormData}
+                isDisabled={!selectedCourse}
+              />
+
+              <InputApply
+                title="Full name"
+                value={formData?.full_name}
+                onChange={(evt) => {
+                  setFormData((it) => {
+                    return { ...it, full_name: evt.target.value };
+                  });
+                }}
+              />
+
+              <InputApply
+                title="E-mail"
+                placeholder="email@email.com"
+                value={formData?.email}
+                onChange={(evt) => {
+                  setFormData((it) => {
+                    return { ...it, email: evt.target.value };
+                  });
+                }}
+              />
+
+              <InputApply
+                title="Phone number"
+                placeholder="(___)___-__-__"
+                value={formData?.phone}
+                onChange={(evt) => {
+                  setFormData((it) => {
+                    return { ...it, phone: evt.target.value };
+                  });
+                }}
+              />
+            </div>
+
+            <Button
+              title="Apply"
+              loading={loading}
+              disabled={Object.values(formData).some(
+                (x) => x === null || x === ""
+              )}
+              classes="course-item-btn"
+              type="button"
+              onClick={() => submitForm()}
+            />
+          </section>
+
+          <ImageBackground
+            className="apply--content__right"
+            image={artWorkImg}
+            alt="Art Work"
+          />
+        </>
+      )}
+    </div>
+  );
+};
+
+const mapStateToProps = createStructuredSelector({
+  courses: selectCourses,
+});
+
+const mapDispatchToProps = (dispatch) => ({
```

Comment:
use can use `useDispatch` hook

using `mapDispatchToProps` and `mapStateToProps` it more related to class component

Reply:
make that changes for all places in your project